pkg/deploy/strategy/rolling/rolling.go

```diff
@@ -111,7 +111,7 @@ func NewRollingDeploymentStrategy(namespace string, client kclient.Interface, co
 			},
 		},
 		getUpdateAcceptor: func(timeout time.Duration) kubectl.UpdateAcceptor {
-			return stratsupport.NewFirstContainerReady(client, timeout, NewFirstContainerReadyInterval)
+			return stratsupport.NewOnePodReady(client, timeout, NewFirstContainerReadyInterval)
```

Comment:
It may not be enough to support one pod.  Imagine you have 100 pods and a
percent of 10.  The first pod completes (canary), then the second 10
start.  Only the first goes to ready.  The second 10 start.  Only the first
goes to ready, but by this time only 5 of the original 20 have started.  At
this point, only 85% of the pods are running.  Does a user want this?

The algorithm probably has to be that each chunk you grab has to be above
the percentage you specified as you change interval.  Ie, if you have 90
out of 100 ready and 10% as your interval, you need to wait.

On Jul 15, 2015, at 8:59 AM, Dan Mace <notifications@github.com> wrote:

In pkg/deploy/strategy/rolling/rolling.go
<https://github.com/openshift/origin/pull/3701#discussion_r34674822>:

> @@ -111,7 +111,7 @@ func NewRollingDeploymentStrategy(namespace string, client kclient.Interface, co
>  			},
>  		},
>  		getUpdateAcceptor: func(timeout time.Duration) kubectl.UpdateAcceptor {
> -			return stratsupport.NewFirstContainerReady(client, timeout, NewFirstContainerReadyInterval)
> +			return stratsupport.NewOnePodReady(client, timeout, NewFirstContainerReadyInterval)

OnePodReadyPerInterval?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3701/files#r34674822>.